src/api/routes/bns/names.ts

```diff
@@ -147,26 +329,21 @@ export function createBnsNamesRouter(db: PgStore, chainId: ChainID): express.Rou
               zonefile: result.zonefile,
               zonefile_hash: result.zonefile_hash,
             };
+            const response = Object.fromEntries(
+              Object.entries(nameInfoResponse).filter(([_, v]) => v != null)
+            ) as typeof nameInfoResponse;
+            await reply.send(response);
           }
-
-          const response = Object.fromEntries(
-            Object.entries(nameInfoResponse).filter(([_, v]) => v != null)
-          );
-          return response;
         })
-        .then(response => {
-          setETagCacheHeaders(res);
-          res.json(response);
-        })
-        .catch(error => {
+        .catch(async error => {
           if (error instanceof NameRedirectError) {
-            res.redirect(error.message);
+            await reply.redirect(error.message);
           } else {
-            res.status(404).json(error);
+            await reply.status(404).send(error);
           }
         });
-    })
+    }
   );
 
-  return router;
-}
+  await Promise.resolve();
```

Comment:
To avoid having to do this, you could declare all these route plugins as `FastifyPluginCallback` (instead of `Async`) and just call `done()` here at the end

Reply:
I'm a fan of the async one because the type checker / eslint enforces the correct behavior. With `FastifyPluginCallback`, the `done()` is easily omitted and then there's just a mysterious timeout at runtime.